### PR TITLE
MEPTS-702 | Adds melhoria de qualidade categoria 12  Part 2 indicators

### PR DESCRIPTION
**Pull request by Edgencio**

### CODE REVIEW

**Comment by ningosi** [comment]
Hi @Edgencio please resolve the conflicts on this PR

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java [hunk 1]
```diff
@@ -882,6 +882,137 @@ public CohortDefinition getMQC11DEN(String indicatorFlag) {
...
+    CohortDefinition b1E =
+            getPatientsFromFichaClinicaB1OrB2("B1E"); // B1E
+
+    CohortDefinition b2 =
+            getPatientsFromFichaClinicaB1OrB2("B2_12"); // B2
+
+    CohortDefinition b2E =
+            getPatientsFromFichaClinicaB1OrB2("B2E"); // B2
+
+    // CohortDefinition nutritionalClass = getPatientsWithNutritionalState();
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java]: If this is NOT being used just delete it

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java [hunk 2]
```diff
@@ -882,6 +882,137 @@ public CohortDefinition getMQC11DEN(String indicatorFlag) {
...
+
+    compositionCohortDefinition.addSearch("C", EptsReportUtils.map(pregnant, MAPPING));
+
+    compositionCohortDefinition.addSearch("D", EptsReportUtils.map(breastfeeding, MAPPING));
+
+    compositionCohortDefinition.addSearch("E", EptsReportUtils.map(transferIn, MAPPING));
+
+    compositionCohortDefinition.addSearch("F", EptsReportUtils.map(transfOut, MAPPING));
+
+    if (indicatorFlag == "A" || indicatorFlag == "E" || indicatorFlag == "F")
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java]: String values are compared using '==', not 'equals()', please replace **==** with **equals()**

Comment (ningosi) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java]: Do that for all where there is string comparisons, only use == for non string values

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java [hunk 3]
```diff
@@ -882,6 +882,137 @@ public CohortDefinition getMQC11DEN(String indicatorFlag) {
...
+                    "once",
+                    hivMetadata.getMasterCardEncounterType(),
+                    commonMetadata.getTransferFromOtherFacilityConcept(),
+                    Collections.singletonList(hivMetadata.getYesConcept()),
+                    hivMetadata.getTypeOfPatientTransferredFrom(),
+                    Collections.singletonList(hivMetadata.getArtStatus()));
+
+    CohortDefinition transfOut = commonCohortQueries.getTranferredOutPatients();
+
+    compositionCohortDefinition.addSearch("A", EptsReportUtils.map(startedART, MAPPING));
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java]: The mappings applied here looks NOT matching the specs, I read the following on the Specs document

Comment (ningosi) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java]: Is it NOT suppossed to be **"startDate=${endDate-14m},endDate=${endDate-11m},location=${location}"**?

**Review by ningosi — COMMENTED** [review]
Look at this mappings

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java [hunk 4]
```diff
@@ -882,6 +882,137 @@ public CohortDefinition getMQC11DEN(String indicatorFlag) {
...
+
+    compositionCohortDefinition.addSearch(
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java]: Here you need to adjust the mappings as the specs says, some how our mappings need to subtract 14 and 11 months from end date then be applied as start and end dates respectively

**Review by ningosi — COMMENTED** [review]
@Edgencio looks like the age applied on these indicators are based on **endDate** the specs says **Age should be calculated on Patient ART Start Date** please check

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java (hunk 4 above)
Comment (Edgencio) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java]: well noted... Thank you !

**Review by ningosi — APPROVED** [review]
Builds and runs fine

**Comment by ningosi** [comment]
@Edgencio good Job, merging and transitioning this, I will base on it to get the numerator values

**Comment by Edgencio** [comment]
Perfect...

**Pull request merged**